File: go/hashmaps/majority_element_test.go
package hashmaps

import "testing"

func TestMajorityElement(t *testing.T) {
	tests := []struct {
		name     string
		nums     []int
		expected int
	}{
		{"basic", []int{3, 2, 3}, 3},
		{"single", []int{1}, 1},
		{"large majority", []int{2, 2, 1, 1, 1, 2, 2}, 2},
		{"clear majority", []int{1, 1, 1, 2}, 1},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			result := MajorityElement(tt.nums)
			if result != tt.expected {
				t.Errorf("got %v, want %v", result, tt.expected)
			}
		})
	}
}

File: typescript/hashmaps/majorityElement.test.ts
import { majorityElement } from './majorityElement';

describe('majorityElement', () => {
  it('should find majority element in basic case', () => {
    expect(majorityElement([3, 2, 3])).toBe(3);
  });

  it('should handle single element', () => {
    expect(majorityElement([1])).toBe(1);
  });

  it('should find majority in larger array', () => {
    expect(majorityElement([2, 2, 1, 1, 1, 2, 2])).toBe(2);
  });

  it('should handle clear majority', () => {
    expect(majorityElement([1, 1, 1, 2])).toBe(1);
  });
});
